src/CoreMetadata/CoreMetadataHeader.jsx

```diff
@@ -44,6 +48,15 @@ function canUserDownload(user, projectAvail, projectID) {
 }
 
 class CoreMetadataHeader extends Component {
+  onGenerateSignedURL = () => {
+    this.props.onGenerateSignedURL(this.props.metadata.object_id);
+  };
+
+  onSignedURLPopupClose = () => {
+    this.props.onUpdatePopup({ signedURLPopup: false });
+    this.props.onClearSignedURL();
```

Comment:
🤔 I don't understand here why do we need to clear url?

Reply:
I was just thinking to be secure? So if u close the popup then you lose it

but might be an overkill...

Reply:
i see... maybe we don't need to worry about that :p up to you